requirements/common.in

```diff
@@ -79,7 +79,7 @@ httplib2==0.12.1
 hypchat==0.21
 
 # For doing highly usable Python profiling
-line_profiler==2.1.2
+-e git+https://github.com/rkern/line_profiler.git@df8dfc88049effeac5fb77830bc9456052ea5c33#egg=line_profiler
```

Comment:
So, what I'd like to happen here is we open an issue with `rkern/line_profiler` asking them to do a release, presenting the traceback one gets without the upgrade on Debian 10 Buster, and mentioning it's already fixed in master (I assume that commit just the latest master?).  Then we can put a comment with a link to the issue, which will make it easy for us to remember that we can clean this up once that release happens.

Also should be `egg=line_profiler==2.1.2.zulip1` at the end of the line.

Reply:
Also I forked it as https://github.com/zulip/line_profiler -- we always use zulip/ repos in our requirements files so that we have control over what happens.

(Possibly, we should add a lint rule for this -- check that there's no `github.com/(anything but zulip)`)

Reply:
> So, what I'd like to happen here is we open an issue with `rkern/line_profiler` asking them to do a release, presenting the traceback one gets without the upgrade on Debian 10 Buster, and mentioning it's already fixed in master (I assume that commit just the latest master?). Then we can put a comment with a link to the issue, which will make it easy for us to remember that we can clean this up once that release happens.
> 
> Also should be `egg=line_profiler==2.1.2.zulip1` at the end of the line.

So the commit is latest master but the issue is really about how it is packaged. It uses the precompiled c files in the pypi package which causes it to break with python3.7 . There is already a [PR](https://github.com/rkern/line_profiler/pull/149) for this. We can just hold on and wait for the next release of line_profiler or we can use this workaround

Reply:
The project is not super active and unlikely to do a release in the near future (maintainer had a baby in February, it sounds like).  I think the best option would be to include that PR in our fork and then just build off that -- you can test easily by making your own fork.

(and then, if I'm not mistaken, that should automatically do the right thing with Cython?)

Reply:
> The project is not super active and unlikely to do a release in the near future (maintainer had a baby in February, it sounds like). I think the best option would be to include that PR in our fork and then just build off that -- you can test easily by making your own fork.
> 
> (and then, if I'm not mistaken, that should automatically do the right thing with Cython?)

It should work fine with that PR although I haven't tested it yet

Reply:
Yeah, please test that.  I also emailed the maintainers asking if we can help in any way.

Reply:
I tested it with the PR mentioned above and it works fine. We can use that but you'll have to merge that PR in the zulip/line_profiler fork.

Reply:
I merged that PR to the zulip fork.